Sources/AppStoreConnectCLI/Commands/TestFlight/BetaTesters/CreateBetaTesterCommand.swift

```diff
@@ -7,80 +7,85 @@ import Foundation
-            .create(betaTesterWithEmail: self.email, firstName: self.firstName, lastName: self.lastName, betaGroupIds: ids)
-        }
-
-        switch (buildIds, groupNames) {
-            case (let buildIds, _) where !buildIds.isEmpty:
-                let endpoint = createWithBuildIds(buildIds)
-                request = api.request(endpoint).eraseToAnyPublisher()
-
-            case (_, let groupNames) where !groupNames.isEmpty:
-                let endpoint = APIEndpoint.betaGroups(filter: [ListBetaGroups.Filter.name(groupNames)])
-
-                let groupIds = api.request(endpoint).map { $0.data.map(\.id) }
-
-                request = groupIds
-                    .flatMap { (groupIds: [String]) -> AnyPublisher<BetaTesterResponse, Error> in
-                        guard !groupIds.isEmpty else {
-                            return Fail(error: CommandError.couldntFindBetaGroup).eraseToAnyPublisher()
-                        }
+        _ = api
+            // Find app resource id matching bundle Id
+            .appResourceId(matching: bundleId)
+            // Find beta groups matching app resource Id
+            .flatMap {
+                api.request(APIEndpoint.betaGroups(forAppWithId: $0)).eraseToAnyPublisher()
+            }
+            // Check if input group names are belong to the app, else throws error
+            .tryMap { [groups, bundleId] (response: BetaGroupsResponse) -> AnyPublisher<[String], Error> in
+                let groupNamesInApp = Set(response.data.compactMap { $0.attributes?.name })
+                let inputGroupNames = Set(groups)
 
-                        let endpoint = createWithGroupIds(groupIds)
+                guard inputGroupNames.isSubset(of: groupNamesInApp) else {
+                    throw CommandError.noGroupsExist(groupNames: groups, bundleId: bundleId)
+                }
 
-                        return api.request(endpoint).eraseToAnyPublisher()
-                    }
+                return try api.betaGroupIdentifiers(matching: groups)
+            }
+            .flatMap { $0 }
+            // Invite tester to the input groups
+            .flatMap { [email, firstName, lastName] (groupIds: [String]) -> AnyPublisher<BetaTesterResponse, Error> in
+                // A tester can only be invite to one group at a time
+                let requests = groupIds.map { (groupId: String) -> AnyPublisher<BetaTesterResponse, Error> in
+                    let endpoint = APIEndpoint.create(betaTesterWithEmail: email,
+                                                      firstName: firstName,
+                                                      lastName: lastName,
+                                                      betaGroupIds: [groupId])
+                    return api.request(endpoint).eraseToAnyPublisher()
+                }
+
+                return Publishers.ConcatenateMany(requests)
+                    .last()
```

Comment:
What does giving us the last result give us. Why not `collect` ?

Reply:
@adamjcampbell A tester can only be invited to one group at a time, so I create multiple invitations to the same tester.
All of the response are pointing to the same tester, I only need the tester's id to get info, so I use the last() to pick one of the results. 
Or should I use `.collect().last`?